apps/minaintyg/src/pages/Certificates/components/CertificateListFilter/CertificateListFilter.test.tsx

```diff
@@ -0,0 +1,92 @@
+import { fakeCertificate, fakeHSA, faker, fakerFromSchema } from '@frontend/fake'
+import { render, screen, waitFor, within } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { rest } from 'msw'
+import { Provider } from 'react-redux'
+import { vi } from 'vitest'
+import { server } from '../../../../mocks/server'
+import { CertificateStatusEnum } from '../../../../schema/certificateList.schema'
+import { certificateFilterOptionsSchema } from '../../../../schema/certificateListFilter.schema'
+import { store } from '../../../../store/store'
+import { CertificateListFilter } from './CertificateListFilter'
+
+const options = {
+  statuses: CertificateStatusEnum.options,
+  certificateTypes: faker.helpers.uniqueArray(fakeCertificate, 4).map(({ id, label }) => ({ id, name: label })),
+  units: faker.helpers.uniqueArray(() => ({ id: fakeHSA(), name: faker.company.name() }), 4),
+  years: ['2023', '2022', '2021', '2020'],
+}
+
+beforeEach(() => {
+  server.use(
+    rest.get('/api/certificate/filters', (_, res, ctx) =>
+      res(ctx.status(200), ctx.json(fakerFromSchema(certificateFilterOptionsSchema)(options)))
+    )
+  )
+})
+
+describe('Options from API', () => {
+  it.each(['Status', 'Mottagning', 'Intygstyp', 'År'])('Should have corret %s option', async (name) => {
+    const { container } = render(
+      <Provider store={store}>
```

Comment:
Can be extracted into common function to reduce code in each test and make them easier to read